### PR TITLE
Add pipeline to release LocalStack tagged OpenAPI spec

### PR DESCRIPTION
**Pull request by giograno**
With the introduction of the LocalStack OpenAPI spec, we should release a tagged spec along side the global LocalStack release.

This PR introduces a new workflow to implement this. In a nutshell:

- We first install the LocalStack packaged we just released;
- We then call the python script that creates the new artifact, i.e., a `yaml` file with the tagged spec  (it required the last version of such script from https://github.com/localstack/openapi/pull/6);
- We create a release commit and a draft release;

This workflow will be dispatched from the "Release" pipeline with the `peter-evans/repository-dispatch@v3` action, similar to the workflow updating the latest spec we already have in this repo.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          bin/release-helper.sh pip-download-retry localstack-core ${{ env.release }}
+          bin/release-helper.sh pip-download-retry localstack-ext ${{ env.release }}
+
+      - name: "Install LocalStack and LocalStack-ext"
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+          pip install localstack==${{ env.release }}
+          pip install localstack-core==${{ env.release }}
+          pip install localstack-ext==${{ env.release }}
```
Comment (alexrashed) [inline .github/workflows/release.yml:56]: nit: Actually, this shouldn't be necessary, as they are direct and exactly pinned dependencies of `localstack`. However, they also don't do any harm.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - name: "Create tagged LocalStack OpenAPI spec"
+        run: |
+          source .venv/bin/activate
+          pip install click
+          pip install pyyaml
+          python bin/update-aws-spec.py
+
+      - name: "Commit release version"
+        run: |
+          DEPENDENCY_FILE="openapi/" bin/release-helper.sh git-commit-release ${{ env.release }}
```
Comment (alexrashed) [inline .github/workflows/release.yml:67]: Q: What does this `DEPENDENCY_FILE="openapi/"` do? This is a folder, and not a `pyproject.toml` or a `setup.cfg`... Would definitely be worth a comment I guess 😛

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+env:
+  git_user_name: localstack[bot]
+  git_user_email: localstack-bot@users.noreply.github.com
+
+jobs:
+  release-localstack-openapi:
+    runs-on: buildjet-2vcpu-ubuntu-2204
+    env:
+      release: ${{ github.event_name == 'workflow_dispatch' && inputs.releaseVersion || github.event.client_payload.version}}
```
Comment (alexrashed) [inline .github/workflows/release.yml:20]: nit: Would it make sense to have the same `client_payload` input name as defined in the `workflow_dispatch`? And is the `client_payload` maybe even already part of the inputs context? In this case you would be able to really just directly use it? :)

**Review by alexrashed — APPROVED** [review]
Nice! Great to see that the OpenAPI specs are officially integrated into the release process! 🚀 🤖 💯

**Comment by giograno** [comment]
Thanks for the review @alexrashed. https://github.com/localstack/openapi/pull/6 needs to also be reviewed and merged for this workflow to work as expected.

**Pull request merged**